### ISSUE TITLE
A question about the finetuning few shot configuration regarding the FsDet preperation: "Which annotation file to provide is not clear"

### ISSUE THREAD

**Issue by muratbayrktr** [LiewFeng/imTED#16]
I have a custom dataset with following summary:
 |  class   |   images   boxes      |
│ car    │   8144 │  188839 │
│ person │   9228 │139964│
│ Total  │   9946 |    328803 │

There are 2 base classes in my dataset. Later I prepared another class with 355 images and 2000~ boxes and merged this into my main dataset. Then I prepared my split using the [prepare_coco_few_shot.py](https://github.com/ucbdrive/few-shot-object-detection/blob/master/datasets/prepare_coco_few_shot.py). And have the exact format as mentioned for my 3 classes.

Training:
I trained my model without the 3rd (novel class) [namely I had 2 classes while training] using the`configs/imted/imted_faster_rcnn_vit_base_mae_3x_coco.py` and obtained a pretrained model let's call it X. 
I proceeded to finetune my model on the few shot task for the novel classes. Created a configuration file containing all 3 classes with the novel one being the last. I am confused about how to fill the part below in the few shot configuration file:

```
pretrained = "path/to/the/X.pth"
.....

classes =  ('truck', 'traffic light', 'fire hydrant', 'stop sign', 'parking meter', 'bench', 'elephant', 'bear', 'zebra', 'giraffe', 'backpack', 'umbrella', 
'handbag', 'tie', 'suitcase', 'frisbee', 'skis', 'snowboard', 'sports ball', 'kite', 'baseball bat', 'baseball glove', 'skateboard', 'surfboard', 
'tennis racket', 'wine glass', 'cup', 'fork', 'knife', 'spoon', 'bowl', 'banana', 'apple', 'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 
'pizza', 'donut', 'cake', 'bed', 'toilet', 'laptop', 'mouse', 'remote', 'keyboard', 'cell phone', 'microwave', 'oven', 'toaster', 'sink', 'refrigerator', 
'book', 'clock', 'vase', 'scissors', 'teddy bear', 'hair drier', 'toothbrush', 'person', 'bicycle', 'car', 'motorcycle', 'airplane', 'bus', 'train', 'boat', 
'bird', 'cat', 'dog', 'horse', 'sheep', 'cow', 'bottle', 'chair', 'couch', 'potted plant', 'dining table', 'tv')
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        pipeline=train_pipeline,
        classes = classes, 
        ann_file='path/of/annotations/json'),
    val=dict(
        classes = classes, 
        ann_file='path/of/annotations/json'),
    test=dict(
        classes = classes, 
        ann_file='path/of/annotations/json'))
```

I did the below configuration at first: 
```
classes =  ('person', 'car', 'mynovel')
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        pipeline=train_pipeline,
        classes = classes, 
        ann_file= 'data/cocosplit/seed1/full_box_30shot_mynovel_trainval.json',
        img_prefix='data/merged_dataset/all/'), 
    val=dict(
        classes = classes, 
        ann_file='data/merged_dataset/annotations/fsod_test.json',
        img_prefix='data/merged_dataset/all/'), 
    test=dict(
        classes = classes, 
        ann_file= 'data/merged_dataset/annotations/fsod_test.json', 
        img_prefix='data/merged_dataset/all/'))  
```

However, the results I am getting at epoch 102 is:
```
Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.001
Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=1000 ] = 0.004
Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=1000 ] = 0.000
Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.000
Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.000
Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.003
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.043
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=300 ] = 0.043
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=1000 ] = 0.043
Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.003
Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.030
Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.120
```
Note that, my prior classes (car and person) got around 0.724 AP score for above metrics.

Also I left load_from = None as I didn't know what to put there because I only got one pretrained model and I am training the model that I can load right now.

I also considered training with the fsod_train but I have more than 30 shots for the novel class there. I checked the original trainvalno5k.json and it's also the same. 

To summarize, I am having problems writing the configuration file for the few shot. I think, how you utilize the datasplit/{trainvalno5k, 5k}.json  and the novel shots (the json files under the cocosplit/seeds) is ambiguous and would be very happy if you could shed a light on this topic.

**Comment by LiewFeng** [comment]
Hi, @muratbayrktr, sorry for the late reply. You may try to modify 60 to 2 [here](https://github.com/LiewFeng/imTED/blob/fe960cccd0357f2915df2da68ac42f4abce14668/mmdet/apis/train.py#L102). For the reason, you may refer to [here](https://github.com/LiewFeng/imTED/issues/15#issuecomment-1667581633).

**Comment by LiewFeng** [comment]
And, the weight for pretrained should be downloaded from [here](https://mailsucasaccn-my.sharepoint.com/:u:/g/personal/liufeng20_mails_ucas_ac_cn/ET7GIhcqLY1KlqfsYthBrU4BLf1BwgboqXwTALxdxHKlUA?e=Z7aOuC) or [here](https://dl.fbaipublicfiles.com/mae/pretrain/mae_pretrain_vit_base_full.pth) or [here](https://dl.fbaipublicfiles.com/mae/pretrain/mae_pretrain_vit_large_full.pth). The 'X' should be the path for load_from.

**Comment by muratbayrktr** [comment]
> Hi, @muratbayrktr, sorry for the late reply. You may try to modify 60 to 2 [here](https://github.com/LiewFeng/imTED/blob/fe960cccd0357f2915df2da68ac42f4abce14668/mmdet/apis/train.py#L102). For the reason, you may refer to [here](https://github.com/LiewFeng/imTED/issues/15#issuecomment-1667581633).

Hi thank you for the response. Having this changed, fixed the issue. Also I looked up the definition of load_from and pretained from the mmdet issues [here](https://github.com/open-mmlab/mmdetection/issues/1247). Knowing the difference helped me fix the confusion too.
